core/tile-entity/inventory.model.lua

```diff
@@ -0,0 +1,43 @@
+local component = require("component")
+local tileEntity = require("core.tile-entity")
+
+---@class inventory: tileEntity
+---@field name string
+---@field size number
+local inventory = {}
+
+---Creates a new inventory object
+---@param address string
+---@param location coordinates
+---@return inventory
+function inventory.new(address, location)
+    ---@type inventory
+    local self = tileEntity.new(address, location)
+
+    local proxy = component.proxy(address)
+    self.name = proxy.getInventoryName()
```

Comment:
This seems wrong, inventories by themselves do not usually have a component with an address, instead they are accessed using the address of a transposer and a side.

Reply:
🤔